### PR TITLE
refactor(VBottomSheet): convert to TypeScript

### PR DESCRIPTION
**Pull request by sh7dm**
## Description
Convert VBottomSheet to TypeScript.

## Motivation and Context
TS conversion.

## How Has This Been Tested?
Markup + `jest`

## Markup:
<details>

```vue
<template>
  <div class="text-xs-center">
    <v-bottom-sheet v-model="sheet">
      <v-btn
        slot="activator"
        color="purple"
        dark
      >
        Click me
      </v-btn>

      <v-list>
        <v-subheader>Open in</v-subheader>
        <v-list-tile
          v-for="tile in tiles"
          :key="tile.title"
          @click="sheet = false"
        >
          <v-list-tile-avatar>
            <v-avatar size="32px" tile>
              <img
                :src="`https://cdn.vuetifyjs.com/images/bottom-sheets/${tile.img}`"
                :alt="tile.title"
              >
            </v-avatar>
          </v-list-tile-avatar>
          <v-list-tile-title>{{ tile.title }}</v-list-tile-title>
        </v-list-tile>
      </v-list>
    </v-bottom-sheet>
  </div>
</template>

<script>
  export default {
    data: () => ({
      sheet: false,
      tiles: [
        { img: 'keep.png', title: 'Keep' },
        { img: 'inbox.png', title: 'Inbox' },
        { img: 'hangouts.png', title: 'Hangouts' },
        { img: 'messenger.png', title: 'Messenger' },
        { img: 'google.png', title: 'Google+' }
      ]
    })
  }
</script>
```
</details>

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VBottomSheet/VBottomSheet.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import '../../stylus/components/_bottom-sheets.styl'
 
 import VDialog from '../VDialog/VDialog'
+import Vue, { VNode } from 'vue'
 
 /* @vue/component */
```
Comment (KaelWD) [inline packages/vuetify/src/components/VBottomSheet/VBottomSheet.ts]: remove

File: packages/vuetify/src/components/VBottomSheet/VBottomSheet.ts [hunk 2]
```diff
@@ -17,10 +18,10 @@ export default {
       default: 'auto'
     },
     persistent: Boolean,
-    value: null
+    value: Boolean
```
Comment (KaelWD) [inline packages/vuetify/src/components/VBottomSheet/VBottomSheet.ts]: I can't find the issue but there was a fair bit of fuss at one point about being able to bind any type to this.

File: packages/vuetify/src/components/VBottomSheet/VBottomSheet.ts [hunk 3]
```diff
@@ -17,10 +18,10 @@ export default {
...
   },
 
-  render (h) {
+  render (h: Function): VNode {
```
Comment (KaelWD) [inline packages/vuetify/src/components/VBottomSheet/VBottomSheet.ts]: Don't annotate `h`

**Comment by johnleider** [comment]
Closing due to inactivity. Please recreate as a branch on the repo.

**Pull request closed without merging**